plugin/src/main/kotlin/com/github/triplet/gradle/play/ModifyReleaseTask.kt

```diff
@@ -0,0 +1,43 @@
+package com.github.triplet.gradle.play
+
+import com.github.triplet.gradle.play.internal.PlayPublishPackageBase
+import com.github.triplet.gradle.play.internal.TrackType
+import org.gradle.api.tasks.TaskAction
+
+open class ModifyReleaseTask : PlayPublishPackageBase() {
+    init {
+        // Always out-of-date since we don't know what's changed on the network
+        outputs.upToDateWhen { false }
+    }
+
+    private var startCodes: List<Long>? = null
+
+    @TaskAction
+    fun modify() {
+        progressLogger.start("Modifies release tracks for variant ${variant.name}", null)
+
+        requireNotNull(extension._fromTrack) {
+            progressLogger.completed("No from-track provided to modify", true)
+        }
+        downloadVersionCodes(extension._fromTrack!!)
+
+        startCodes.let {
+            if (it == null || it.isEmpty()) {
+                progressLogger.completed("No versions published on track '${extension.track}' for ${variant.name}", false)
```

Comment:
The Gradle API is kinda bad. Your final status won't be visible because the task will finish immediately afterwards. Basically, what I'm saying is that there should only be one `completed` call without a status at the end of the method.

Reply:
Interesting, seems like a poor design.